src/main/java/seedu/address/model/recipe/Ingredient.java

```diff
@@ -43,20 +37,26 @@
      *
      * @param name A valid name.
      * @param quantity A valid quantity.
-     * @param quantifier A valid quantifier.
+     * @param quantifier A valid quantifier, blanks are allowed.
      */
     public Ingredient(String name, double quantity, String quantifier) {
         requireNonNull(name);
         checkArgument(isValidIngredientName(name), NAME_CONSTRAINTS);
 
         checkArgument(isValidQuantity(quantity), QUANTITY_CONSTRAINTS);
 
-        requireNonNull(quantifier);
-        checkArgument(isValidQuantifier(quantifier), QUANTIFIER_CONSTRAINTS);
+
+        if (quantifier == null) {
+            quantifier = "";
+        }
```

Comment:
Is this null check necessary? I think the only times this constructor is called is when a proper quantifier is parsed, so don't think it'll ever be null, since during instances where user doesn't set a quantifier the other constructor will be called

Reply:
I think it's a viable option (only) for testing to cover such cases